### PR TITLE
Add tests for Algorithms section

### PR DESCRIPTION
**Pull request by marcoscaceres**
This covers the [algorithms](https://w3c.github.io/payment-request/#algorithms) section - but excludes the accepts right now. 

*  Shipping address changed algorithm
* Shipping option changed algorithm
* User aborts the payment request algorithm

The following is called indirectly: 

 * PaymentRequest updated algorithm

The "User accepts the payment request algorithm" is tested elsewhere.

### CODE REVIEW

**Comment by marcoscaceres** [comment]
@rsolomakhin, in the API Chrome seems to have a bug whereby if you register an event handler first (e.g., `.onshippingaddresschange`), the listeners get ignored? If, however, you add the listeners first (using `.addEventListener()`), then they all work. This tests actually triggers the bug.

**Comment by w3c-bots** [comment]
# Build PASSED

Started: 2017-08-28 09:45:09
Finished: 2017-08-28 09:54:03


<h2>Failing Jobs</h2>
<ul>

<li>safari:10.0</li>

<li>MicrosoftEdge:14.14393</li>

</ul>




View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/17820)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/269125559)

**Comment by rsolomakhin** [comment]
@marcoscaceres: Thank you, I'll fix.

File: payment-request/algorithms-manual.https.html [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+setup({
+  explicit_done: true,
+  explicit_timeout: true,
+});
+const methods = [
+  {
+    supportedMethods: ["basic-card"],
```
Comment (edenchuang) [inline payment-request/algorithms-manual.https.html]: s/["basic-card"]/"basic-card"

File: payment-request/algorithms-manual.https.html [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+      request.addEventListener(eventName, event => {
+        events.push(event);
+        resolve("listener-2");
+      });
+    });
+    try {
+      const response = await request.show();
+      response.complete("success");
+    } catch (err) {
+      await request.abort();
```
Comment (edenchuang) [inline payment-request/algorithms-manual.https.html]: I guess no need to call "await request.abort()", request.show() should execute about() internally once there is any error during showing. Wouldn't "await request.abort()" cause another errors? such as "InvalidStateError".

Comment (marcoscaceres) [reply payment-request/algorithms-manual.https.html]: yeah, good point. I'll remove.

File: payment-request/user-abort-algorithm-manual.https.html [hunk 3]
```diff
@@ -0,0 +1,62 @@
+<!doctype html>
+<meta charset="utf8">
+<link rel="help" href="https://w3c.github.io/payment-request/#dom-paymentresponse-complete()">
```
Comment (edenchuang) [inline payment-request/user-abort-algorithm-manual.https.html]: I think the link should be https://w3c.github.io/payment-request/#user-aborts-the-payment-request-algorithm.

File: payment-request/user-abort-algorithm-manual.https.html [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+<title>
+  PaymentResponse.prototype.complete() method
```
Comment (edenchuang) [inline payment-request/user-abort-algorithm-manual.https.html]: should be changed to User aborts the payment request algorithm

**Review by edenchuang — APPROVED** [review]
Please change the link and title to fit the test.

**Pull request merged**